pytorch_lightning/plugins/precision/precision_plugin.py

```diff
@@ -119,24 +114,25 @@ def clip_gradients(
         if clip_val <= 0:
             return
 
-        clip_grad_func = self.clip_grad_funcs[gradient_clip_algorithm]
-        clip_grad_func(optimizer, clip_val)  # type: ignore
+        if gradient_clip_algorithm == GradClipAlgorithmType.VALUE:
+            self.clip_grad_by_value(optimizer, clip_val)
+        elif gradient_clip_algorithm == GradClipAlgorithmType.NORM:
+            # TODO: there should be a mechanism to set `norm_type`
+            self.clip_grad_by_norm(optimizer, clip_val, eps=self.EPSILON)
```

Comment:
this is different here, are you sure?

Reply:
I couldn't get mypy to work with the previous way of pulling the function out of the dict.

I think having `if-elif` is simpler and okay